src/table/columnCheckboxes.component.tsx

```diff
@@ -1,55 +1,128 @@
-      ? column.channelInfo?.userFriendlyName
-      : column.channelInfo?.systemName
-      ? column.channelInfo?.systemName
-      : accessor;
-    return accessor && accessor.toUpperCase() !== 'TIMESTAMP' ? (
-      <div key={accessor}>
-        <label htmlFor={accessor}>{label}</label>
-        <Checkbox
-          onChange={(e) => handleColumnChecked(e.target.id, e.target.checked)}
-          id={accessor}
-          value={accessor}
-          checked={shouldBeChecked(accessor)}
-          inputProps={{
-            'aria-label': `${accessor} checkbox`,
-          }}
-        />
-      </div>
-    ) : null;
+const ColumnCheckboxes = React.memo((): React.ReactElement => {
+  const { data: availableColumns } = useAvailableColumns();
+
+  const selectableColumns = React.useMemo(
+    () =>
+      availableColumns?.filter((col) => {
+        const accessor = col.accessor?.toString();
+        return accessor && accessor.toUpperCase() !== 'TIMESTAMP';
+      }) ?? [],
+    [availableColumns]
+  );
+
+  const selectedColumns = useAppSelector((state) =>
+    selectSelectedColumns(state, availableColumns ?? [])
+  );
+  const dispatch = useAppDispatch();
+
+  const onColumnOpen = React.useCallback(
+    (column: string): void => {
+      dispatch(selectColumn(column));
+    },
+    [dispatch]
+  );
+
+  const onColumnClose = React.useCallback(
+    (column: string): void => {
+      dispatch(deselectColumn(column));
+    },
+    [dispatch]
+  );
+
+  const handleColumnChecked = React.useCallback(
+    (accessor: string, checked: boolean) => {
+      checked ? onColumnOpen(accessor) : onColumnClose(accessor);
+    },
+    [onColumnClose, onColumnOpen]
+  );
+
+  const shouldBeChecked = React.useCallback(
+    (columnAccessor: string): boolean => {
+      const match = selectedColumns.filter((col: Column) => {
+        return col.accessor === columnAccessor;
+      });
+
+      return match && match.length > 0;
+    },
+    [selectedColumns]
+  );
+
+  const parentRef = React.useRef<HTMLDivElement>(null);
+
+  const rowVirtualizer = useVirtualizer({
+    count: selectableColumns.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 42,
+    overscan: 20,
   });
 
-  return <div style={{ overflow: 'auto', height: '150px' }}>{checkboxes}</div>;
-};
+  const CheckboxRow = React.useCallback(
+    (props: { column: Column }) => {
+      const { column } = props;
+      const accessor = column.accessor?.toString();
+      const label = column.channelInfo?.userFriendlyName
+        ? column.channelInfo?.userFriendlyName
+        : column.channelInfo?.systemName
+        ? column.channelInfo?.systemName
+        : accessor;
+      return (
+        <>
+          <label htmlFor={accessor}>{label}</label>
+          <Checkbox
+            onChange={(e) => handleColumnChecked(e.target.id, e.target.checked)}
+            id={accessor}
+            value={accessor}
+            checked={shouldBeChecked(accessor ?? '')}
```

Comment:
Loving the changes so far! I'm keeping an eye on what's being done so I can provide a timely review when ready.

Regarding this line, out of curiosity, is it more efficient to instead say
```
checked={accessor ? shouldBeChecked(accessor) : false}
```
so that time isn't spent running through `selectedColumns` looking for an empty string?. That being said, I can't think of a realistic scenario when `accessor` isn't undefined.